server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java

```diff
@@ -207,8 +206,10 @@ private SearchResults searchData(DbSession dbSession, SearchProjectsRequest requ
   }
 
   @CheckForNull
-  private Set<String> buildFilterOnFavoriteProjectUuids(List<String> criteria, Set<String> favoriteProjectUuids) {
-    if (hasIsFavoriteCriterion(criteria)) {
+  private Set<String> buildFilterOnFavoriteProjectUuids(List<FilterParser.Criterion> criteria, Set<String> favoriteProjectUuids) {
```

Comment:
I'm really surprised by the API. It shouldn't return null or the parameter. It should return a boolean and then be handled in the caller method – I don't say you created it